src/main/scala/com/amazon/deequ/anomalydetection/RelativeRateOfChangeStrategy.scala

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ * use this file except in compliance with the License. A copy of the License
+ * is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.deequ.anomalydetection
+
+import breeze.linalg.DenseVector
+
+/**
+  * Detects anomalies based on the values' rate of change.
+  * The order of the difference can be set manually.
+  * If it is set to 0, this strategy acts like the [[SimpleThresholdStrategy]].
+  *
+  * RelativeRateOfChangeStrategy(Some(0.9), Some(1.1), 1) for example
+  * calculates the first discrete difference
+  * and if some point's value changes by more than 10.0 Percent in one timestep,
+  * it flags it as an anomaly.
+  *
+  * @param maxRateDecrease Lower bound of accepted relative change (as new value / old value).
+  * @param maxRateIncrease Upper bound of accepted relative change (as new value / old value).
+  * @param order           Order of the calculated difference.
+  *                        Set to 1 it calculates the difference between two consecutive values.
+  */
+case class RelativeRateOfChangeStrategy(
+    maxRateDecrease: Option[Double] = None,
+    maxRateIncrease: Option[Double] = None,
+    order: Int = 1)
+  extends BaseChangeStrategy {
```

Comment:
AFAIK, inheritance and case classes do not go together well. Can we change BaseChangeStrategy to an object and just call its methods?

Reply:
I can change it to a trait. I think that's the better way.